### ISSUE TITLE
Integrate with the Windows 11 Context Menu

### ISSUE THREAD

**Issue by xandfis** [microsoft/vscode#127365]
The new Windows 11 context menu uses IExplorerCommand + app identity for app integration. Identity is established for unpackaged Win32 like VS Code using Sparse Manifests. 

A sample can be found here: https://github.com/microsoft/AppModelSamples/tree/master/Samples/SparsePackages

The context menu extension is useful for a few key reasons:

1. File Type associations don't work with extensionless files
2. A context menu verb is the only way to open a whole folder of files in VS Code
3. File Type Associations can be an extra set of steps when opening a file is a one-off action

![image](https://user-images.githubusercontent.com/69924000/123696073-30848680-d810-11eb-96c0-1abc7ffb9820.png)

**Comment by sbatten** [comment]
@joaomoreno need to test myself but this seems to be effectively a bug if this feature is broken in win11

**Comment by xandfis** [comment]
@sbatten Is there a way to convert this to a bug from a feature ask, or should I file a new one?

**Comment by sbatten** [comment]
I have add the bug label

**Comment by isidorn** [comment]
@sbatten I have talked to @xandfis over teams, feel free to ping me for more details if needed.

**Comment by joaomoreno** [comment]
> @joaomoreno need to test myself but this seems to be effectively a bug if this feature is broken in win11

Yup, definitely a bug on Windows if that's the case.

**Comment by Bramgus12** [comment]
Its actually still in the context menu, but you have to click on `show more options`. This will show the old windows 10 context menu. But I think it would still be better if it is on the first page next to windows terminal.

**Comment by sbatten** [comment]
@joaomoreno i think its not necessarily a bug in Windows but a change in API if I understand OP correctly. I'll ping @isidorn for more info.

**Comment by isidorn** [comment]
Let's first wait for this change to be done https://github.com/microsoft/vscode/issues/127075
And then double check the Win11 experience after the default vscode install (associated to file types)

**Comment by dstrome** [comment]
@isidorn Why does this need to be dependent on #127075? That work isn't scheduled until August and is just fit and finish. This issue is significantly impacting my work flow. It may not seem like much but I use this option with extreme frequency. Please reconsider.

**Comment by joaomoreno** [comment]
> significantly impacting my work flow

Nothing about Windows 11 should significantly impact anyone's workflow today. The fact that you're on Windows 11 means you enrolled in the Insider program, Dev channel. Please have patience.

**Comment by dstrome** [comment]
@joaomoreno  It shouldn't, but it is. I'm on dev/internal because we've been encouraged to selfhost. So do I need to choose between working and selfhosting?

**Comment by joaomoreno** [comment]
Tell me why again you are forced to choose? The feature you want is there, just one additional click away. It's called a workaround, something you should expect when selfhosting on dev previews.

**Comment by JustinGrote** [comment]
@dstrome  if you've been encouraged to use Windows 11 by your company then your company accepts the risks of using an insider preview, this is one of them.

I'm also on Windows 11 and while I find it annoying to have to click once more, it's not really a massive, breaking change to my workflow, and I can patiently wait until this gets implemented. If it's really that big of a problem for you, you can submit a PR for it yourself :)

**Comment by siddagra** [comment]
@JustinGrote  That does not work anymore

**Comment by JustinGrote** [comment]
> @JustinGrote  That does not work anymore

Brilliant insight....

**Comment by MafiaBar** [comment]
> The new Windows 11 context menu uses IExplorerCommand + app identity for app integration. Identity is established for unpackaged Win32 like VS Code using Sparse Manifests.
> 
> A sample can be found here: https://github.com/microsoft/AppModelSamples/tree/master/Samples/SparsePackages
> 
> The context menu extension is useful for a few key reasons:
> 
> 1. File Type associations don't work with extensionless files
> 2. A context menu verb is the only way to open a whole folder of files in VS Code
> 3. File Type Associations can be an extra set of steps when opening a file is a one-off action
> 
> ![image](https://user-images.githubusercontent.com/69924000/123696073-30848680-d810-11eb-96c0-1abc7ffb9820.png)

How We Can Get This Done ?

**Comment by siddagra** [comment]
I ended up using an ahk script as a temporary fix. 
Autohotkey: https://www.autohotkey.com/

Script:
```
#IfWinActive ahk_class CabinetWClass
RButton::
Click, Left
Send {Shift Down}
Send {F10}
Send {Shift Up}
```

**Comment by MafiaBar** [comment]
> I ended up using an ahk script as a temporary fix.
> Autohotkey: https://www.autohotkey.com/
> 
> Script:
> 
> ```
> #IfWinActive ahk_class CabinetWClass
> RButton::
> Click, Left
> Send {Shift Down}
> Send {F10}
> Send {Shift Up}
> ```
> 
> Image search may be a better solution though, as left and then right-clicking on a file/folder will cause you to open that file/folder with this script.

Bro Simply Bring The Old Menu Context Back 
https://gadgetstouse.com/blog/2021/07/16/get-the-old-right-click-context-menu-on-windows-11/
Why Would you use it Like That

**Comment by siddagra** [comment]
> > I ended up using an ahk script as a temporary fix.
> > Autohotkey: https://www.autohotkey.com/
> > Script:
> > ```
> > #IfWinActive ahk_class CabinetWClass
> > RButton::
> > Click, Left
> > Send {Shift Down}
> > Send {F10}
> > Send {Shift Up}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Image search may be a better solution though, as left and then right-clicking on a file/folder will cause you to open that file/folder with this script.
> 
> Bro Simply Bring The Old Menu Context Back
> https://gadgetstouse.com/blog/2021/07/16/get-the-old-right-click-context-menu-on-windows-11/
> Why Would you use it Like That

I would if it worked, but it doesn't, so I can't.

**Comment by JustinGrote** [comment]
@shivan they probably implemented iexplorer command like I mentioned above

**Comment by imbushuo** [comment]
I've attempted to implement a small POC to surface some potential blockers before making a formal feature idea or pull request. Code is currently a mess so I will try to get it somewhere later.

![image](https://user-images.githubusercontent.com/6284353/136694489-5a0a5cf7-02ef-47b9-a694-d2d4c51d078a.png)

Issues I've surfaced before going to make it some more formal work:

- The new modern context menu needs a package identity and Windows will read some assets from it. I assume VS Code will not be fully MSIX packaged at this moment? But even with sparse package, Windows will want some extra image assets (mainly icon files), otherwise the icon on taskbar when launching via modern context menu will be weird. I use what currently have in `resources`, but looks like the manifest will need some extra care:

![image](https://user-images.githubusercontent.com/6284353/136694866-4752fc34-5df4-49a6-a307-44c775028ac6.png)

- How will the sparse package gets registered and unregistered? Along with the installer? 
- Since the modern context menu is implemented using in-proc COM server (this is the only supported approach), the population logic is implemented in code. Then how will it read the user preference of context menu enablement? I've think of letting it read `HKCR` (the old key such as `Computer\HKEY_CLASSES_ROOT\*\shell\VSCode\command`) to determine if user have enabled context menu display, but this feels slightly hacky.

**Comment by MafiaBar** [comment]
> I've attempted to implement a small POC to surface some potential blockers before making a formal feature idea or pull request. Code is currently a mess so I will try to get it somewhere later.
> 
> ![image](https://user-images.githubusercontent.com/6284353/136694489-5a0a5cf7-02ef-47b9-a694-d2d4c51d078a.png)
> 
> Issues I've surfaced before going to make it some more formal work:
> 
> * The new modern context menu needs a package identity and Windows will read some assets from it. I assume VS Code will not be fully MSIX packaged at this moment? But even with sparse package, Windows will want some extra image assets (mainly icon files), otherwise the icon on taskbar when launching via modern context menu will be weird. I use what currently have in `resources`, but looks like the manifest will need some extra care:
> 
> ![image](https://user-images.githubusercontent.com/6284353/136694866-4752fc34-5df4-49a6-a307-44c775028ac6.png)
> 
> * How will the sparse package gets registered and unregistered? Along with the installer?
> * Since the modern context menu is implemented using in-proc COM server (this is the only supported approach), the population logic is implemented in code. Then how will it read the user preference of context menu enablement? I've think of letting it read `HKCR` (the old key such as `Computer\HKEY_CLASSES_ROOT\*\shell\VSCode\command`) to determine if user have enabled context menu display, but this feels slightly hacky.

Can you share the Code With Me ?

**Comment by xandfis** [comment]
@MafiaBar 
Please check these out:

- Blog post: https://blogs.windows.com/windowsdeveloper/2021/07/19/extending-the-context-menu-and-share-dialog-in-windows-11/
- Sample code: https://github.com/microsoft/AppModelSamples/tree/master/Samples/SparsePackages

**Comment by MafiaBar** [comment]
> Hi @imbushuo can you share your codes? also winrar was able to add items to the windows 11 context menu, Winrar is not released as a packaged, but it was able to do so. https://www.win-rar.com/singlenewsview.html?&L=0&tx_ttnews%5Btt_news%5D=175&cHash=aca38142108c76995eba6c674bfd0e14

They are already shared the source code: https://github.com/microsoft/AppModelSamples/tree/master/Samples/SparsePackages

**Comment by ghost1372** [comment]
@JustinGrote  @MafiaBar  A solution in an [article](https://www.howtogeek.com/howto/windows-vista/add-take-ownership-to-explorer-right-click-menu-in-vista/) written in 2017 can easily create a key in the right-click menu.
But it seems to only work on the folder and run as an admin
![Untitled](https://user-images.githubusercontent.com/9213496/137632898-064c1b41-7b25-4b38-bfd1-f5f09643bbb7.png)

**Comment by MafiaBar** [comment]
> @JustinGrote @MafiaBar A solution in an [article](https://www.howtogeek.com/howto/windows-vista/add-take-ownership-to-explorer-right-click-menu-in-vista/) written in 2017 can easily create a key in the right-click menu. But it seems to only work on the folder and run as an admin ![Untitled](https://user-images.githubusercontent.com/9213496/137632898-064c1b41-7b25-4b38-bfd1-f5f09643bbb7.png)

Bro This Only Works For Take Ownership and even it doesn't use IExplorer Command And Sparse Package

**Comment by OzWookiee** [comment]
I didn't realise how much I used something until I moved to W11. Looking forward to this update!

**Comment by LoKy-dev** [comment]
> I didn't realise how much I used something until I moved to W11. Looking forward to this update!

You only realize what you need whe you lose it 🥲

**Comment by YoraiLevi** [comment]
Why is this not in already?

**Comment by mikeldoka1** [comment]
yeah why not already

**Comment by MafiaBar** [comment]
I still can't make it work. Come on Microsoft give us  a  clean documentation instead of a complicated example with C#/C++

**Comment by egoward** [comment]
Pretty bumpy to get it setup and I don't know 'touchwp' is, but this app can be setup to add an entry as expected -
  https://www.microsoft.com/store/productId/9PC7BZZ28G0X
Maybe someone in MS can find 'touchwp' and nudge them towards careers.microsoft.com :)

**Comment by MafiaBar** [comment]
This Looks cool, I'll take a look at it through the weekend

**Comment by moudey** [comment]
Shell provides functions and expressions through which you can fully control the customization of the context menu by adding new menu items or modifying system menu items. Supports all system files including virtual folders and taskbar.
More details at this link: https://nilesoft.org/shell

![Shell](https://nilesoft.org/images/screenshots/ss3.png)
![Shell](https://nilesoft.org/images/screenshots/ss5.png)

**Comment by JustinGrote** [comment]
@moudey can you explain how that's relevant here? This only works with the classic windows menu, not windows 11.

**Comment by moudey** [comment]
> @moudey can you explain how that's relevant here? This only works with the classic windows menu, not windows 11.

Shell currently only supports the legacy context menu but is under development to support the modern context menu.

**Comment by MafiaBar** [comment]
For now it doesn't support + these kind of application will make your Window slower

**Comment by isidorn** [comment]
Hi all 👋 
I have commented on the PR, but let me also comment here for transparency.

While we acknowledge that the PR fixes the issue it would introduce a dependency on native code which we would like to avoid. The Windows team has confirmed to us that native code is required here and that adding "Open with Code" is no longer supported using the registry keys.

Though we are exploring alternative approaches to this which could potentially reduce the complexity. We will have more details in the next couple of weeks. I will keep you posted here. Due to that assigning this to the February milestone. Thank you!

**Comment by AveYo** [comment]
> Hi all 👋 I have commented on the PR, but let me also comment here for transparency.
> 
> While we acknowledge that the PR fixes the issue it would introduce a dependency on native code which we would like to avoid. The Windows team has confirmed to us that native code is required here and that adding "Open with Code" is no longer supported using the registry keys.
> 
> We are exploring alternative approaches to this which could potentially reduce the complexity. We will have more details in the next couple of weeks. I will keep you posted here. Due to that assigning this to the February milestone. Thank you!

**Hint**: https://github.com/AveYo/LeanAndMean/blob/main/RunAsTI.reg
Whitelisted ids can be found in C:\Windows\SystemApps\MicrosoftWindows.Client.CBS_cw5n1h2txyewy\FileExplorerExtensions.dll i.e. _Windows.CompressTo, addnetworkplace, connectNetworkDrive, disconnectNetworkDrive_ etc. 
With some clever picks and use of AppliesTo filters can easily accommodate vscode ;)

**Comment by coverxit** [comment]
> > Hi all 👋 I have commented on the PR, but let me also comment here for transparency.
> > While we acknowledge that the PR fixes the issue it would introduce a dependency on native code which we would like to avoid. The Windows team has confirmed to us that native code is required here and that adding "Open with Code" is no longer supported using the registry keys.
> > We are exploring alternative approaches to this which could potentially reduce the complexity. We will have more details in the next couple of weeks. I will keep you posted here. Due to that assigning this to the February milestone. Thank you!
> 
> **Hint**: https://github.com/AveYo/LeanAndMean/blob/main/RunAsTI.reg Whitelisted ids can be found in C:\Windows\SystemApps\MicrosoftWindows.Client.CBS_cw5n1h2txyewy\FileExplorerExtensions.dll i.e. _Windows.CompressTo, addnetworkplace, connectNetworkDrive, disconnectNetworkDrive_ etc. With some clever picks and use of AppliesTo filters can easily accommodate vscode ;)

Wow, thanks for your finding! I was able to do a dirty change in the registry until official support is there 👀 

![image](https://user-images.githubusercontent.com/1834970/152139451-998910d6-99eb-4ab7-9766-65c702e7649d.png)

**Comment by MafiaBar** [comment]
How did you do that @coverxit  ?
Can you explain everything please ?

**Comment by Masternova** [comment]
> > > Hi all 👋 I have commented on the PR, but let me also comment here for transparency.
> > > While we acknowledge that the PR fixes the issue it would introduce a dependency on native code which we would like to avoid. The Windows team has confirmed to us that native code is required here and that adding "Open with Code" is no longer supported using the registry keys.
> > > We are exploring alternative approaches to this which could potentially reduce the complexity. We will have more details in the next couple of weeks. I will keep you posted here. Due to that assigning this to the February milestone. Thank you!
> > 
> > 
> > **Hint**: https://github.com/AveYo/LeanAndMean/blob/main/RunAsTI.reg Whitelisted ids can be found in C:\Windows\SystemApps\MicrosoftWindows.Client.CBS_cw5n1h2txyewy\FileExplorerExtensions.dll i.e. _Windows.CompressTo, addnetworkplace, connectNetworkDrive, disconnectNetworkDrive_ etc. With some clever picks and use of AppliesTo filters can easily accommodate vscode ;)
> 
> Wow, thanks for your finding! I was able to do a dirty patch in the registry until official support is there 👀
> 
> ![image](https://user-images.githubusercontent.com/1834970/152139451-998910d6-99eb-4ab7-9766-65c702e7649d.png)

Nice! Could you share how you did this?

**Comment by coverxit** [comment]
@MafiaBar @Masternova 

Use at your own risk. Remember to change `<username>` in the path accordingly.

```
Windows Registry Editor Version 5.00

[HKEY_CLASSES_ROOT\*\shell\pintohome]
"MUIVerb"="Edit with Code"

[HKEY_CLASSES_ROOT\*\shell\pintohome\command]
@="\"C:\\Users\\<username>\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe\" \"%1\""
```

**Comment by amlxv** [comment]
> 

Does it work for the folders?

**Comment by Masternova** [comment]
> > @MafiaBar @Masternova
> > Use at your own risk. Remember to change `<username>` in the path accordingly.
> > ```
> > Windows Registry Editor Version 5.00
> > 
> > [HKEY_CLASSES_ROOT\*\shell\pintohome]
> > "MUIVerb"="Edit with Code"
> > 
> > [HKEY_CLASSES_ROOT\*\shell\pintohome\command]
> > @="\"C:\\Users\\<username>\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe\" \"%1\""
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > As @AveYo mentioned, you can use one of the whitelisted ids in FileExplorerExtensions.dll to achieve this. Here I'm using `pintohome`.
> > The one in the screenshot is `opennewwindow`. But somehow it conflicts with pinned apps in taskbar and they end up being opened in VS Code... I tried to add `AppliesTo` on `.lnk` but still doesn't work.
> > So I switched to `pintohome` and it seems okay at least for now.
> 
> Does it also work for the folders? I tried once but it didn't appear.

I was able to get it to work with folders by doing the same thing in the registry for pintohome for folders

HKEY_CLASSES_ROOT\Folder\shell\pintohome change the MUIVerb to "Open in Code"
HKEY_CLASSES_ROOT\Folder\shell\pintohome\command change the (Default) to "C:\Program Files\Microsoft VS Code\Code.exe" "%1"

change the location Code.exe to wherever you installed yours. I also changed DelegateExecute to DelegateExecute-old and MUIVerb to MUIVerb-old so that I could revert these changes later if needed. 

I am not well versed in making .reg files or I would just write out some code to do this automatically for you if I could

**Comment by AngusWarren** [comment]
Thanks! This is working well for me with files and folders.

```PowerShell
$VSCodeExe = "$env:LOCALAPPDATA\Programs\Microsoft VS Code\Code.exe"
if (-not (Test-Path $VSCodeExe)) {
    $VSCodeExe = "$env:ProgramFiles\Microsoft VS Code\Code.exe"
}
'*','Folder' | ForEach-Object {
    $Path = 'HKCU:\Software\Classes\{0}\shell\pintohome' -f $_
    $null = New-Item -Path "$Path\command" -Force
    Set-ItemProperty -LiteralPath $Path -Name 'MUIVerb' -Value 'Open with VS Code'
    Set-ItemProperty -LiteralPath "$Path\command" -Name '(Default)' -Value ('"{0}" "%1"' -f $VSCodeExe)
    Remove-ItemProperty -LiteralPath "$Path\command" -Name "DelegateExecute" -ErrorAction SilentlyContinue
}
```

**Comment by tommai78101** [comment]
@coverxit 

Was wondering if this is considered as "a method using the Windows native dependency (registry keys) that VS Code wanted to avoid doing"?

**Comment by sredna** [comment]
> What was said about "Windows native dependency" is vague for me.

Nobody except you has mentioned "Windows native dependency" (whatever that means). What has been mentioned is "native code" which means a non-interpreted language, usually C/C++. This is because the supported implementation for context menus now require everyone to write their own shell extension for some insane reason.

**Comment by coverxit** [comment]
> @coverxit 
> 
> 
> 
> Was wondering if this is considered as "a method using the Windows native dependency (registry keys) that VS Code wanted to avoid doing"?
> 
> 
> 
> What was said about "Windows native dependency" is vague for me.

Like I said this was a dirty patch, the real problem is that `pintohome` is indeed meant for "Pin to quick access" and you can't really change the icon, otherwise the icon of "Pin to quick access" will be changed as well.

Also this is something like  undocumented APIs, so it might stop working if Windows dropped it someday.

**Comment by tommai78101** [comment]
@sredna 

> Nobody except you has mentioned "Windows native dependency" (whatever that means). What has been mentioned is "native code" which means a non-interpreted language, usually C/C++. This is because the supported implementation for context menus now require everyone to write their own shell extension for some insane reason.

I got it wrong, sorry. I was referring to this "dependency on native code", which I misinterpret it as "Windows native dependency" or something.

> While we acknowledge that the PR fixes the issue it would introduce a dependency on native code which we would like to avoid.

Which I guess is what you mentioned about writing their own shell extension.

**Comment by AveYo** [comment]
@coverxit that's just an unfortunate verb choice. `install`, `extract`, `runas` would fit better without any downside since we're talking code files or folders for which those verbs would not do anything by default. Like I said when I introduced it, there are enough alternatives, and can further filter via AppliesTo entry.

**Comment by coverxit** [comment]
> @coverxit that's just an unfortunate verb choice. `install`, `extract`, `runas` would fit better without any downside since we're talking code files or folders for which those verbs would not do anything by default. Like I said when I introduced it, there are enough alternatives, and can further filter via AppliesTo entry.

Not sure if this is my own problem or it indeed doesn't work, but I've tried these verbs you mentioned, however none of them are shown up in the context menu.

**Comment by Ldev007** [comment]
> > > @MafiaBar @Masternova
> > > Use at your own risk. Remember to change `<username>` in the path accordingly.
> > > ```
> > > Windows Registry Editor Version 5.00
> > > 
> > > [HKEY_CLASSES_ROOT\*\shell\pintohome]
> > > "MUIVerb"="Edit with Code"
> > > 
> > > [HKEY_CLASSES_ROOT\*\shell\pintohome\command]
> > > @="\"C:\\Users\\<username>\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe\" \"%1\""
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > As @AveYo mentioned, you can use one of the whitelisted ids in FileExplorerExtensions.dll to achieve this. Here I'm using `pintohome`.
> > > The one in the screenshot is `opennewwindow`. But somehow it conflicts with pinned apps in taskbar and they end up being opened in VS Code... I tried to add `AppliesTo` on `.lnk` but still doesn't work.
> > > So I switched to `pintohome` and it seems okay at least for now.
> > 
> > 
> > Does it also work for the folders? I tried once but it didn't appear.
> 
> I was able to get it to work with folders by doing the same thing in the registry for pintohome for folders
> 
> HKEY_CLASSES_ROOT\Folder\shell\pintohome change the MUIVerb to "Open in Code" HKEY_CLASSES_ROOT\Folder\shell\pintohome\command change the (Default) to "C:\Program Files\Microsoft VS Code\Code.exe" "%1"
> 
> change the location Code.exe to wherever you installed yours. I also changed DelegateExecute to DelegateExecute-old and MUIVerb to MUIVerb-old so that I could revert these changes later if needed.
> 
> I am not well versed in making .reg files or I would just write out some code to do this automatically for you if I could

@Masternova This one didn't work out for me right away. I had to do some more tweaking to make it work. Anyways, thanks for the hint though ! it got me to the right place and @coverxit you made me do the right things :^)

**Comment by RehngruppenEF** [comment]
> 

```
Windows Registry Editor Version 5.00`

[HKEY_CLASSES_ROOT\Folder\shell\pintohome]
"MUIVerb"="Edit with Code"

[HKEY_CLASSES_ROOT\Folder\shell\pintohome\command]
@="\"C:\\Users\\<username>\\AppData\\Local\\Programs\\Microsoft VS Code\\Code.exe\" \"-n\" \"%1\""
```

Note the extra "-n" on the path to open it in a new window

**Comment by hatton** [comment]
Like some other folks here, I couldn't get it to work at first. Solved by removing the existing DelegateExecute. AngusWarren's Powerscript script above does this, but for some reason, it was still there for me.

**Comment by VidarDev** [comment]
It's funny that winrar can do it but Microsoft's own VSCode can't get it right. Pls fix that.

<img width="532" alt="Screenshot 2022-02-22 181804" src="https://user-images.githubusercontent.com/98586665/155194217-8f16d971-3ebc-4a7b-be93-8f2d25087196.png">

**Comment by MattBDev** [comment]
I hope this makes the cut for the March 2022 milestone and doesn't get pushed back again.

**Comment by ghost1372** [comment]
> It's funny that winrar can do it but Microsoft's own VSCode can't get it right. Pls fix that.
> 
> <img width="532" alt="Screenshot 2022-02-22 181804" src="https://user-images.githubusercontent.com/98586665/155194217-8f16d971-3ebc-4a7b-be93-8f2d25087196.png">
> 

just for information, Winrar is written with C++ (The new menu can be easily implemented) but VSCode is written with electron!

**Comment by VidarDev** [comment]
> > It's funny that winrar can do it but Microsoft's own VSCode can't get it right. Pls fix that.
> > 
> > <img width="532" alt="Screenshot 2022-02-22 181804" src="https://user-images.githubusercontent.com/98586665/155194217-8f16d971-3ebc-4a7b-be93-8f2d25087196.png">
> > 
> 
> just for information, Winrar is written with C++ (The new menu can be easily implemented) but VSCode is written with electron! 

just for Information, WinRar is a small Organisation. And VS Code is Windows so you should trust a company like that to coordinate their programs.

**Comment by VidarDev** [comment]
> > It's funny that winrar can do it but Microsoft's own VSCode can't get it right. Pls fix that.
> > 
> > <img width="532" alt="Screenshot 2022-02-22 181804" src="https://user-images.githubusercontent.com/98586665/155194217-8f16d971-3ebc-4a7b-be93-8f2d25087196.png">
> > 
> 
> just for information, Winrar is written with C++ (The new menu can be easily implemented) but VSCode is written with electron! 

just for Information, WinRar is a small Organisation. And VS Code is Windows so you should trust a company like that to coordinate their programs.

**Comment by Diegorro98** [comment]
> > > It's funny that winrar can do it but Microsoft's own VSCode can't get it right. Pls fix that.
> > > <img alt="Screenshot 2022-02-22 181804" width="532" src="https://user-images.githubusercontent.com/98586665/155194217-8f16d971-3ebc-4a7b-be93-8f2d25087196.png">
> > 
> > 
> > just for information, Winrar is written with C++ (The new menu can be easily implemented) but VSCode is written with electron!
> 
> just for Information, WinRar is a small Organisation. And VS Code is Windows so you should trust a company like that to coordinate their programs.

@VidarDev It is true that VS Code belongs to Microsoft, but VS Code is not based on native code because one of the VS Code principles is to be aviable in al platforms and even in the web, to achive this VS Code uses electron, which is a multiplatform web based technology (if I'm wrong, please, correct me).
Using this technology implies sometimes not be able to use some native features (or wait until it's supported on electron, like [mica effect](https://github.com/electron/electron/issues/29937#issuecomment-1046008182)). For this feature, just in case you haven't readed the [comments](https://github.com/microsoft/vscode/issues/127365#issuecomment-1016460501) from above, they even asked the Windows team about how the context menu implementation works and they already know how to do it, even a PR exist solving the problem, but the thing is that this solution requires to introduce a dependency on native code, so they want to find an easier solution for this.

**Comment by PeterWone** [comment]
This is not priority work. 
If you don't like the broken Win11 context menu then turn it off.

```
reg delete "HKCU\Software\Classes\CLSID\{86ca1aa0-34aa-4e8b-a509-50c905bae2a2}" /f /ve
```

**Comment by MafiaBar** [comment]
> reg delete "HKCU\Software\Classes\CLSID\{86ca1aa0-34aa-4e8b-a509-50c905bae2a2}" /f /ve

It's not broken at all dude.

**Comment by sredna** [comment]
> It's not broken at all dude

It is working as designed, yes. The problem is that the design made it a point to enforce signing requirements to lock out small developers and user added tweaks. This was supposedly done to "clean up" the menu but if that is what they wanted they could just hide classic registry entries by default and a settings UI to control them. We basically had [this ability](https://docs.microsoft.com/en-us/windows/win32/shell/how-to-define-extended-verbs) since Windows 2000.

This is just going to end up in a silly arms race where most developers will have to create shell extensions or use the hacks, and the menus will get slower because more code has to be loaded and executed.

**Comment by PeterWone** [comment]
> It's not broken at all dude.

* To those who don't like the change in behaviour, it's broken. Rather like the Office ribbon.
* Philosophically I'm in the Raymond Chen camp.
* Now that you know what I meant, let's not debate this.

**Comment by B3ret** [comment]
> Thanks! This is working well for me with files and folders.
> 
> ```powershell
> $VSCodeExe = "$env:LOCALAPPDATA\Programs\Microsoft VS Code\Code.exe"
> if (-not (Test-Path $VSCodeExe)) {
>     $VSCodeExe = "$env:ProgramFiles\Microsoft VS Code\Code.exe"
> }
> '*','Folder' | ForEach-Object {
>     $Path = 'HKCU:\Software\Classes\{0}\shell\pintohome' -f $_
>     $null = New-Item -Path "$Path\command" -Force
>     Set-ItemProperty -LiteralPath $Path -Name 'MUIVerb' -Value 'Open with VS Code'
>     Set-ItemProperty -LiteralPath "$Path\command" -Name '(Default)' -Value ('"{0}" "%1"' -f $VSCodeExe)
>     Remove-ItemProperty -LiteralPath "$Path\command" -Name "DelegateExecute" -ErrorAction SilentlyContinue
> }
> ```

This is decent solution for me, since I don't need "Pin To Home" in the context menu (can just drag and drop). On my version of Windows 11 however, the vscode-"command" within HKCU:\Software\Classes\Folder\shell\pintohome is not changing existing behavior, i.e., pintohome seems hardcoded to always pin to home, even if the name of the entry changes.
I therefore had to add a traditional item that lands in the "more options" for this case, unfortunately. I also added a key to HKCU:\Software\Classes\Directory\shell\pintohome; it works for folders in the left tree part of the explorer.

```
$VSCodeExe = "$env:LOCALAPPDATA\Programs\Microsoft VS Code\Code.exe"
if (-not (Test-Path $VSCodeExe)) {
    $VSCodeExe = "$env:ProgramFiles\Microsoft VS Code\Code.exe"
}
'*', 'Directory' | ForEach-Object {
    $Path = 'HKCU:\Software\Classes\{0}\shell\pintohome' -f $_
    $null = New-Item -Path "$Path\command" -Force
    Set-ItemProperty -LiteralPath $Path -Name 'MUIVerb' -Value '🔷 Open with VS Code'
    Set-ItemProperty -LiteralPath "$Path\command" -Name '(Default)' -Value ('"{0}" -n "%1"' -f $VSCodeExe)
    Remove-ItemProperty -LiteralPath "$Path\command" -Name "DelegateExecute" -ErrorAction SilentlyContinue
}

'Folder' | ForEach-Object {
    $Path = 'HKCU:\Software\Classes\{0}\shell\VsCode' -f $_
    $null = New-Item -Path "$Path\command" -Force
    Set-ItemProperty -LiteralPath $Path -Name '(Default)' -Value '🔷 Open with VS Code'
    Set-ItemProperty -LiteralPath "$Path\command" -Name '(Default)' -Value ('"{0}" -n "%1"' -f $VSCodeExe)
    Remove-ItemProperty -LiteralPath "$Path\command" -Name "DelegateExecute" -ErrorAction SilentlyContinue
}
```

**Comment by MattBDev** [comment]
It's unfortunate that this ended up as a stretch goal for March after being pushed back in February.

**Comment by isaumya** [comment]
Well, @MattBDev it's super unfortunate that this has been pushed again further to April 2022 now. Really great job Microsoft. Considering VS Code a Microsoft product that is taking so much time to implement the Win 11 context menu while third-party apps like WinRAR are doing it already.

This is seriously setting great precedence for the other third-party devs out there. Bravo. 👏👏👏

**Comment by LasseRosenow** [comment]
> Well, @MattBDev it's super unfortunate that this has been pushed again further to April 2022 now. Really great job Microsoft. Considering VS Code a Microsoft product that is taking so much time to implement the Win 11 context menu while third-party apps like WinRAR are doing it already.
> 
> This is seriously setting great precedence for the other third-party devs out there. Bravo. 👏👏👏

Please don't act on it as if it's the end of the world and please stay respectful.

**Comment by fcharlie** [comment]
<img width="858" alt="E152E89E-7C6B-4847-9333-C57622A01C84" src="https://user-images.githubusercontent.com/6904176/160944130-786593cd-83c6-46c7-90c4-a7f1f394bf0b.png">

**Comment by AveYo** [comment]
**Why is this still an issue and people upset about it when they can fix it themselves?**  

Just not over-complicate things _(kept saying use a better verb)_ - here, let me do it for you:  
```ps1
ren -lit HKCU:\Software\Classes\Drive\shell\VSCode extract
ren -lit HKCU:\Software\Classes\Directory\Background\shell\VSCode extract
ren -lit HKCU:\Software\Classes\Directory\shell\VSCode extract
ren -lit HKCU:\Software\Classes\*\shell\VSCode extract
sp -lit HKCU:\Software\Classes\*\shell\extract "AppliesTo" "NOT .zip"
```
or to undo:
```.ps1
ren -lit HKCU:\Software\Classes\Drive\shell\extract VSCode
ren -lit HKCU:\Software\Classes\Directory\Background\shell\extract VSCode
ren -lit HKCU:\Software\Classes\Directory\shell\extract VSCode
ren -lit HKCU:\Software\Classes\*\shell\extract VSCode
rp -lit HKCU:\Software\Classes\*\shell\VSCode "AppliesTo"
```

Obviously installing as usual with the default context menu entries, 
Then post install do the above in terminal or powershell..  

As for maintainers, since windows 11 hides user entries, **hijacking entries back (smartly) is fair game**, so just offer a choice in the install wizard for something like the above

**Comment by phuze** [comment]
As demonstrated by [fcharlie](https://github.com/fcharlie), you can add VS Code to the context menu using [this solution](https://github.com/microsoft/vscode/pull/139640) created by [strear](https://github.com/strear). I have provided detailed instructions [here](https://github.com/microsoft/vscode/pull/139640#issuecomment-1086109339), for those interested in trying it.

**Comment by oghaki** [comment]
> > Well, @MattBDev it's super unfortunate that this has been pushed again further to April 2022 now. Really great job Microsoft. Considering VS Code a Microsoft product that is taking so much time to implement the Win 11 context menu while third-party apps like WinRAR are doing it already.
> > This is seriously setting great precedence for the other third-party devs out there. Bravo. 👏👏👏
> 
> Please don't act on it as if it's the end of the world and please stay respectful.
> 
> Complaining over and over again does not make anyone work faster.

I think you may be misinterpreting him, unless you're concerned about offending Microsoft—it sounds like he's genuinely commending third-party devs who have already incorporated the changes or workarounds necessary to implement proper context-menu functionality in Windows 11, while Microsoft still hasn't accomplished the same with their own software.

**Comment by MattBDev** [comment]
> As demonstrated by [fcharlie](https://github.com/fcharlie), you can add VS Code to the context menu using [this solution](https://github.com/microsoft/vscode/pull/139640) created by [strear](https://github.com/strear). I have provided detailed instructions [here](https://github.com/microsoft/vscode/pull/139640#issuecomment-1086109339), for those interested in trying it.

As much as I appreciate the instructions from you and others as to add this back into the Windows 11 context menu, I would like an official supported solution. That way when my company updates my work PC I won't possibly loose this entry.

**Comment by Abhijeet1520** [comment]
Not sure if this was mentioned before but right-clicking while holding Shift opens the old context menu in windows, so that could be used for now.

`Shift + Right Click`

**Comment by xandfis** [comment]
> Not sure if this was mentioned before but right-clicking while holding Shift opens the old context menu in windows, so that could be used for now.
> 
> `Shift + Right Click`

Only in Insider builds, not in Retail.

**Comment by Sima25k** [comment]
> <img alt="E152E89E-7C6B-4847-9333-C57622A01C84" width="858" src="https://user-images.githubusercontent.com/6904176/160944130-786593cd-83c6-46c7-90c4-a7f1f394bf0b.png">
@fcharlie 我能请教您问题吗，我也来自china，beijing，我的sparsepackagedemo 也不生效

**Comment by fcharlie** [comment]
@Sima25k 在我的开源项目 Baulk 中，有相应的例子：
Appx 清单：
https://github.com/baulk/baulk/tree/master/msix/unscrew

shell extension 扩展：
https://github.com/baulk/baulk/tree/master/extension/unscrew-extension

你需要做的：

+ 编写一个 shell 扩展，实现 IExplorerCommand 接口，记得生成一个随机的 UUID。
+ 在 Appx 中添加 FileExplorerContextMenus。
+ 将生成的 dll 和 Appx 清单以及 Assets 拷贝到一个目录。
+ 使用 MakeAppx 打包。
+ 使用 SignTool 签名（注意签名前你可能需要使用 MakeCert，Pvk2Pfx 生成 pfx 私钥）

这里有个 powershell 脚本用户自动化打包构建：

[https://github.com/baulk/baulk/blob/master/msix/makeUnscrewAppx.ps1](https://github.com/baulk/baulk/blob/master/msix/makeUnscrewAppx.ps1)

该方法不依赖 Visual Studio，但如果需要发布到应用商店，证书的问题需要解决，（我这边使用的是自签名临时证书。）

```xml
  <desktop4:Extension Category="windows.fileExplorerContextMenus">
                    <desktop4:FileExplorerContextMenus>
                        <desktop5:ItemType Type="*">
                            <desktop5:Verb Id="OpenUnscrewDev" Clsid="951C66D1-98D1-4007-8AE4-412E985D70DB" />
                        </desktop5:ItemType>
                    </desktop4:FileExplorerContextMenus>
                </desktop4:Extension>
                <com:Extension Category="windows.comServer">
                    <com:ComServer>
                        <com:SurrogateServer DisplayName="Unscrew Shell Extension">
                            <com:Class Id="951C66D1-98D1-4007-8AE4-412E985D70DB" Path="unscrew-extension.dll" ThreadingModel="STA"/>
                        </com:SurrogateServer>
                    </com:ComServer>
                </com:Extension>
```

**Comment by Sima25k** [comment]
您好，十分感谢您能回复我


1、我使用的是win32 应用 + sparse package的方式


2、例子我使用的是微软官方的：
https://github.com/microsoft/AppModelSamples/tree/master/Samples/SparsePackages


3、 我在利用win32应用注册稀疏包后，可以成功运行（任务管理器，详细信息里，可以看到程序报名）
这样是不是意味着 下面的工作已经完成？
使用 MakeAppx 打包。
使用 SignTool 签名
利用稀疏包给 win32 应用增加标识


我将生成的 dll 和 Appx 清单以及 Assets 拷贝到了win32程序的执行目录，双击执行后，可以成功启动且可以看到程序包名，但是contex menu没有生效


不知道，问题可能出在什么地方呢













在 2022-04-27 15:07:32，"Force Charlie" ***@***.***> 写道：

@Sima25k 在我的开源项目 Baulk 中，有相应的例子：
Appx 清单：
https://github.com/baulk/baulk/tree/master/msix/unscrew

shell extension 扩展：
https://github.com/baulk/baulk/tree/master/extension/unscrew-extension

你需要做的：

编写一个 shell 扩展，实现 IExplorerCommand 接口，记得生成一个随机的 UUID。
在 Appx 中添加 FileExplorerContextMenus。
将生成的 dll 和 Appx 清单以及 Assets 拷贝到一个目录。
使用 MakeAppx 打包。
使用 SignTool 签名（注意签名前你可能需要使用 MakeCert，Pvk2Pfx 生成 pfx 私钥）

这里有个 powershell 脚本用户自动化打包构建：

https://github.com/baulk/baulk/blob/master/msix/makeUnscrewAppx.ps1

该方法不依赖 Visual Studio，但如果需要发布到应用商店，证书的问题需要解决，（我这边使用的是自签名临时证书。）

  <desktop4:ExtensionCategory="windows.fileExplorerContextMenus">

                    <desktop4:FileExplorerContextMenus>

                        <desktop5:ItemTypeType="*">

                            <desktop5:VerbId="OpenUnscrewDev"Clsid="951C66D1-98D1-4007-8AE4-412E985D70DB" />

                        </desktop5:ItemType>

                    </desktop4:FileExplorerContextMenus>

                </desktop4:Extension>

                <com:ExtensionCategory="windows.comServer">

                    <com:ComServer>

                        <com:SurrogateServerDisplayName="Unscrew Shell Extension">

                            <com:ClassId="951C66D1-98D1-4007-8AE4-412E985D70DB"Path="unscrew-extension.dll"ThreadingModel="STA"/>

                        </com:SurrogateServer>

                    </com:ComServer>

                </com:Extension>

**Comment by Sima25k** [comment]
> @Sima25k 在我的开源项目 Baulk 中，有相应的例子： Appx 清单： https://github.com/baulk/baulk/tree/master/msix/unscrew
> 
> shell extension 扩展： https://github.com/baulk/baulk/tree/master/extension/unscrew-extension
> 
> 你需要做的：
> 
> * 编写一个 shell 扩展，实现 IExplorerCommand 接口，记得生成一个随机的 UUID。
> * 在 Appx 中添加 FileExplorerContextMenus。
> * 将生成的 dll 和 Appx 清单以及 Assets 拷贝到一个目录。
> * 使用 MakeAppx 打包。
> * 使用 SignTool 签名（注意签名前你可能需要使用 MakeCert，Pvk2Pfx 生成 pfx 私钥）
> 
> 这里有个 powershell 脚本用户自动化打包构建：
> 
> https://github.com/baulk/baulk/blob/master/msix/makeUnscrewAppx.ps1
> 
> 该方法不依赖 Visual Studio，但如果需要发布到应用商店，证书的问题需要解决，（我这边使用的是自签名临时证书。）
> 
> ```
>   <desktop4:Extension Category="windows.fileExplorerContextMenus">
>                     <desktop4:FileExplorerContextMenus>
>                         <desktop5:ItemType Type="*">
>                             <desktop5:Verb Id="OpenUnscrewDev" Clsid="951C66D1-98D1-4007-8AE4-412E985D70DB" />
>                         </desktop5:ItemType>
>                     </desktop4:FileExplorerContextMenus>
>                 </desktop4:Extension>
>                 <com:Extension Category="windows.comServer">
>                     <com:ComServer>
>                         <com:SurrogateServer DisplayName="Unscrew Shell Extension">
>                             <com:Class Id="951C66D1-98D1-4007-8AE4-412E985D70DB" Path="unscrew-extension.dll" ThreadingModel="STA"/>
>                         </com:SurrogateServer>
>                     </com:ComServer>
>                 </com:Extension>
> ```

@fcharlie 您好，十分感谢您能回复我


1、我使用的是win32 应用 + sparse package的方式


2、例子我使用的是微软官方的：
https://github.com/microsoft/AppModelSamples/tree/master/Samples/SparsePackages


3、 我在利用win32应用注册稀疏包后，可以成功运行（任务管理器，详细信息里，可以看到程序报名）
这样是不是意味着 下面的工作已经完成？
使用 MakeAppx 打包。
使用 SignTool 签名
利用稀疏包给 win32 应用增加标识


我将生成的 dll 和 Appx 清单以及 Assets 拷贝到了win32程序的执行目录，双击执行后，可以成功启动且可以看到程序包名，但是contex menu没有生效


不知道，问题可能出在什么地方呢

**Comment by fcharlie** [comment]
> 

稀疏包也需要安装的，你使用 Add-AppxPackage 安装没？安装后，Windows Explorer 会自动加载。

**Comment by Sima25k** [comment]
> > 
> 
> 稀疏包也需要安装的，你使用 Add-AppxPackage 安装没？安装后，Windows Explorer 会自动加载。

通过 AddPackageByUriAsync，双击稀疏包以后，也提示已安装

**Comment by Sima25k** [comment]
> > 
> 
> 稀疏包也需要安装的，你使用 Add-AppxPackage 安装没？安装后，Windows Explorer 会自动加载。
应该是安装成功了，不然win32 应用应该是不会显示 包名，是有什么我没注意到的吗，比如32 /64，一些权限问题？

**Comment by fcharlie** [comment]
> > > 
> > 
> > 
> > 稀疏包也需要安装的，你使用 Add-AppxPackage 安装没？安装后，Windows Explorer 会自动加载。
> > 应该是安装成功了，不然win32 应用应该是不会显示 包名，是有什么我没注意到的吗，比如32 /64，一些权限问题？

你按照我项目里面的例子再试试。

**Comment by Sima25k** [comment]
> > > > 
> > > 
> > > 
> > > 稀疏包也需要安装的，你使用 Add-AppxPackage 安装没？安装后，Windows Explorer 会自动加载。
> > > 应该是安装成功了，不然win32 应用应该是不会显示 包名，是有什么我没注意到的吗，比如32 /64，一些权限问题？
> 
> 你按照我项目里面的例子再试试。

问题解决了，我可能知道，为什么这么多人，contex menu不生效的问题了
1、在AppxManifest.xml配置的路径，都是相对路径，是以打包好的WNSClientDemo.msix文件为基准的相对路径
2、很多人不熟悉微软的打包流程，都会以单独目录存放打好的WNSClientDemo.msix包，对相对路径的理解应该也是以可执行程序作为基准的
所以虽然可以注册成功，但AppxManifest.xml里的文件加载会失败

**Comment by Sima25k** [comment]
> > > > 
> > > 
> > > 
> > > 稀疏包也需要安装的，你使用 Add-AppxPackage 安装没？安装后，Windows Explorer 会自动加载。
> > > 应该是安装成功了，不然win32 应用应该是不会显示 包名，是有什么我没注意到的吗，比如32 /64，一些权限问题？
> 
> 你按照我项目里面的例子再试试。



> > > > > 
> > > > 
> > > > 
> > > > 稀疏包也需要安装的，你使用 Add-AppxPackage 安装没？安装后，Windows Explorer 会自动加载。
> > > > 应该是安装成功了，不然win32 应用应该是不会显示 包名，是有什么我没注意到的吗，比如32 /64，一些权限问题？
> > 
> > 
> > 你按照我项目里面的例子再试试。
> 
> 问题解决了，我可能知道，为什么这么多人，contex menu不生效的问题了 1、在AppxManifest.xml配置的路径，都是相对路径，是以打包好的WNSClientDemo.msix文件为基准的相对路径 2、很多人不熟悉微软的打包流程，都会以单独目录存放打好的WNSClientDemo.msix包，对相对路径的理解应该也是以可执行程序作为基准的 所以虽然可以注册成功，但AppxManifest.xml里的文件加载会失败

win32 调用 AddPackageByUriAsync 时，可以指定资源的位置

**Comment by Caltor** [comment]
> This is not priority work. If you don't like the broken Win11 context menu then turn it off.
> ```
> reg delete "HKCU\Software\Classes\CLSID\{86ca1aa0-34aa-4e8b-a509-50c905bae2a2}" /f /ve
> ```

For the benefit of others, I think that's the undo. To turn off the new menu you need
`reg.exe add "HKCU\Software\Classes\CLSID\{86ca1aa0-34aa-4e8b-a509-50c905bae2a2**}\InprocServer32" /f /ve`

**Comment by archergod** [comment]
> 

Shift + F10 when folder is selected open old menu.

**Comment by Curious-r** [comment]
This issue is frequent caller of the iteration plan.😉

**Comment by phuze** [comment]
> This issue is frequent caller of the iteration plan.😉

Unfortunate -- this was downgraded from "stretch goal", to -- "god knows when".

**Comment by He1loC** [comment]
That shoudn't be a big problem for MS....what are they doing?

**Comment by Sima25k** [comment]
> 
You can put your shell DLL in the position specified by the first parameter of your AddPackageByUriAsync function

**Comment by Sima25k** [comment]
> That shoudn't be a big problem for MS....what are they doing?

You can put your shell DLL in the position specified by the first parameter of your AddPackageByUriAsync function

**Comment by Sima25k** [comment]
> > This issue is frequent caller of the iteration plan.😉
> 
> Unfortunate -- this was downgraded from "stretch goal", to -- "god knows when". Good thing [strear](https://github.com/strear) is out here doing [gods work](https://github.com/microsoft/vscode/pull/139640).

You can put your shell DLL in the position specified by the first parameter of your AddPackageByUriAsync function

**Comment by Curious-r** [comment]
> That shoudn't be a big problem for MS....what are they doing?

VScode group do not want to introduce a dependency on native code, so they are waiting for win11 team provide other solutions.

**Comment by archergod** [comment]
> > That shoudn't be a big problem for MS....what are they doing?
> 
> VScode group do not want to introduce a dependency on native code, so they are waiting for win11 team provide other solutions.

I understand but isn't it this feature is anyways OS dependent. I mean all OS has different way of handling things, and it is not VSCode's feature par say it is, it's installer feature to make it at home for that OS. So there is no harm using OS dependent code ..

**Comment by JustinGrote** [comment]
> > they are waiting for win11 team provide other solutions
> 
> Win11 was changed on purpose to not allow registry-only extensions. I doubt they are going to change their mind. This was a policy decision to block 3rd-party stuff.

Well it's working quite well in this case 😄

**Comment by cinderblock** [comment]
> I understand but isn't it this feature is anyways OS dependent. I mean all OS has different way of handling things, and it is not VSCode's feature par say it is, it's installer feature to make it at home for that OS. So there is no harm using OS dependent code ..

The issue is not "OS dependent code", it's "OS dependent **native** code".

**Comment by MattBDev** [comment]
Endgame is 4 days away. Looks like this is being pushed back **again** 😞

**Comment by Diegorro98** [comment]
It seems that the task for this issue has been finally completed on the Iteration Plan for September 2022!!! 🎉🎉🎉

**Comment by deepak1556** [comment]
Hi all, this iteration we were mostly finishing up the investigation of the last known issues with the context menu and the plan item was mostly for that. We have not merged the changes to our `main` branch yet, the plan is to make the feature available in Insiders for October 2022 and then based on developer feedback we will plan for adding it to stable.

**Comment by isidorn** [comment]
We have resolved this issue and with the latest VS Code insiders this problem is now fixed.
It should also be available in the next VS Code stable that is about to go out in 10 days.

Do try it out and let us know. Thanks a lot!

**Comment by Diegorro98** [comment]
I have tried downloading  VS Code insiders from Microsoft Store, and the option in the contextual menu doesn't appear!
Just to make sure that everything is like it should be, I checked for updates (no new updates were available) and then I uninstalled it and installed it again but this time downloading form the [web](https://code.visualstudio.com/insiders/), I selected the option for the contextual menu and it worked.

Maybe the option is disabled by default and in Microsoft Store, vscode is installed with default settings. but it can't be changed later (Reinstall is required...)

**Comment by joaomoreno** [comment]
### Insiders

This is now available in VS Code Insiders. In order to use it, please make sure to manually reinstall
VS Code Insiders and choose the correct option in the Setup dialog.

![image](https://user-images.githubusercontent.com/22350/197482873-1eb13025-e459-4dbb-af0d-5729ae96340d.png)

### Stable

This will only be available in VS Code once we're confident of the changes, no earlier than the January milestone.

**Comment by isaumya** [comment]
Hi @Diegorro98,
Yah! It's not just the insider, even the sable VS Code has the same thing. If you install it from MS Store you won't see the option for context menu addition. I have already made this mistake and then I basically uninstalled the MS Store version and downloaded from the website. Then you will see the installation guide along with all the options. So, basically don't install VS Code from MS Store.

**Comment by peterdickx** [comment]
> Hi @Diegorro98, Yah! It's not just the insider, even the sable VS Code has the same thing. If you install it from MS Store you won't see the option for context menu addition. I have already made this mistake and then I basically uninstalled the MS Store version and downloaded from the website. Then you will see the installation guide along with all the options. So, basically don't install VS Code from MS Store.

It doesn't seem to work for directories, only for files?

**Comment by dj-fiorex** [comment]
I can confirm that this is working only for files
VsCode Insiders v.1.74

**Comment by llatpic** [comment]
> I can confirm that this is working only for files VsCode Insiders v.1.74

Today it started working again on folders after some weeks. It worked before just day one on installation. ¯\_(ツ)_/¯

**Comment by marwan-nwh** [comment]
I installed it a few days ago (not from the store), and it doesn't work with folders unless I use "show more options". Am I missing something?

**Comment by llatpic** [comment]
> > I can confirm that this is working only for files VsCode Insiders v.1.74
> 
> Today it started working again on folders after some weeks. It worked before just day one on installation. ¯_(ツ)_/¯

Since a couple of days ago  the Insiders context menu it is gone again. I do not know why.

**Comment by eunminlee** [comment]
How come VsCode Insider v 1.74.0 works but VsCode Stable v 1.74.0?

**Comment by MusaSSH** [comment]
> How come VsCode Insider v 1.74.0 works but VsCode Stable v 1.74.0?

https://github.com/microsoft/vscode/issues/127365#issuecomment-1288556108

**Comment by NinelieZ** [comment]
> We have resolved this issue and with the latest VS Code insiders this problem is now fixed. In order to have the VS Code entry in the context menu a reinstall is required so you can choose the right option in the installer.
> 
> It should be available in the December / January VS Code stable. We need this work to be in VS Code Insiders for a couple of milestones to build confidence. Thanks for understanding
> 
> Do try it out in Insiders and let us know. Thanks a lot!

Anyway to do this manually? (As I'm using vscode in portable mode _(:з」∠)_ )
